groupmne/inverse.py

```diff
@@ -7,93 +7,356 @@
+    if method == "lasso":
+        return IndLasso
+    elif method == "relasso":
+        return IndRewLasso
+    elif method == "mtw":
+        return MTW
+    elif method == "remtw":
+        return ReMTW
+    elif method == "dirty":
+        return DirtyModel
+    elif method == "grouplasso":
+        return GroupLasso
+    else:
+        raise ValueError("Method %s not recognized." % method)
+
+
+def _method_to_str(method):
+    if method == "lasso":
+        return "mutar.IndLasso"
+    if method == "relasso":
+        return "mutar.IndRewLasso"
+    elif method == "mtw":
+        return "mutar.MTW"
+    elif method == "remtw":
+        return "mutar.ReMTW"
+    elif method == "dirty":
+        return "mutar.DirtyModel"
+    elif method == "grouplasso":
+        return "mutar.GroupLasso"
+    else:
+        raise ValueError("Method %s not recognized." % method)
+
+
+class InverseOperator(object):
+    """InverseOperator class to represent the inverse problem data.
 
     Parameters
     ----------
-    gains: ndarray, shape (n_subjects, n_channels, n_sources).
-        Forward data, returned by
-        `group_model.compute_gains` or `group_model.compute_inv_data`.
-    M : ndarray, shape (n_subjects, n_channels, n_times)
-        M-EEG data
-    group_info : dict
-        The measurement info.
-    method : str
-        Inverse problem model to use. For now, only "grouplasso" is
-        supported. The group-lasso solver promotes source estimates with
-        overlapping active vertices across subjects. Each time point is
-        treated independently.
-    depth : float in (0, 1)
-        Depth weighting. If 1, no normalization is done.
-    alpha : float in (0, 1)
-        Regularization hyperparameter set as a fraction of
-        alpha_max for which all sources are 0.
-    return_stc : bool, (optional, default True)
-        If true, source estimates are returned as stc objects, array otherwise.
-    time_independent : bool, (optional, default false)
-        If True, each time point is solved independently. By default,
-        the group lasso is applied on the time and subjects axes.
-    n_jobs : int (default 1)
-        The number of CPUs to use in parallel.
-    kwargs : dict
-        additional arguments passed to the solver.
-
-    Returns
-    -------
-    estimates: instance of SourceEstimates | array
-        The estimated sources as an array if `return_stc` is False.
-    log : dict
-        Some info about the convergence.
+    fwds: list of `mne.Forward`.
+        Forward soluton of each subject.
+    noise_covs: list of `mne.Covariance`
+        Noise covariance of each subject.
+    src_ref: instance of `mne.SourceSpaces`.
+        Source space to be used as a reference for the group study. `src_ref`
+        will be morphed to the individual surface of each subject to have
+        aligned sources.
+    ch_type: str.
+        Type of channels to use. Must be one of "grad", "mag" or "eeg".
```

Comment:
there is no reason to restrict things to single channel types if you pass a noise cov. I would just take any available data channel in the evokeds passed.